packages/@ember/-internals/runtime/lib/system/core_object.js

```diff
@@ -266,7 +266,10 @@ class CoreObject {
     }
 
     registerDestructor(self, ensureDestroyCalled, true);
-    registerDestructor(self, () => self.willDestroy());
+    registerDestructor(self, () => {
+      self.willDestroy();
+      sendEvent(self, 'willDestroy');
```

Comment:
I'm curious, why is this added?

Reply:
This corresponds to the listener added to the FrameworkObject and is meant to follow the implementation of the preexisting handling for `init`: https://github.com/emberjs/ember.js/pull/19212/files#diff-9b45b66d303cab2e652ff010d5825e5ca583e9c658f03d1a638acf4305fab295R77-R78